src/interaction/keyframerecording.cpp

```diff
@@ -0,0 +1,415 @@
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <openspace/interaction/keyframerecording.h>
+
+#include <openspace/camera/camera.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/navigation/navigationhandler.h>
+#include <openspace/navigation/orbitalnavigator.h>
+#include <openspace/scene/scene.h>
+#include <openspace/scene/scenegraphnode.h>
+#include <openspace/util/timemanager.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/glm.h>
+#include <algorithm>
+#include <iostream>
+
+#include "keyframerecording_lua.inl"
+
+namespace {
+    constexpr std::string_view _loggerCat = "KeyframeRecording";
+} // namespace
+
+namespace openspace::interaction::keys {
+    // These keys are const char* since nlohmann::json do not support string_view in .at()
+    constexpr const char* Camera = "camera";
+    constexpr const char* Position = "position";
+    constexpr const char* Rotation = "rotation";
+    constexpr const char* Scale = "scale";
+    constexpr const char* FollowFocusNodeRotation = "followFocusNodeRotation";
+    constexpr const char* FocusNode = "focusNode";
+
+    constexpr const char* Timestamp = "timestamp";
+    constexpr const char* Application = "application";
+    constexpr const char* Sequence = "sequence";
+    constexpr const char* Simulation = "simulation";
+    constexpr const char* X = "x";
+    constexpr const char* Y = "y";
+    constexpr const char* Z = "z";
+    constexpr const char* W = "w";
+}
+
+namespace openspace::interaction {
+
+void to_json(nlohmann::json& j, const KeyframeRecording::Keyframe& keyframe) {
+    j = {
```

Comment:
How about splitting this one up into multiple smaller json objects and then combining them at the end? 

It current